packages/bitkeep/src/adapter.ts

```diff
@@ -0,0 +1,166 @@
+    }
+
+    get connected(): boolean {
+        return !!this._wallet;
+    }
+
+    get autoApprove(): boolean {
+        return false;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            this._connecting = true;
+
+            const wallet = typeof window !== 'undefined' && window.solana;
+            if (!wallet) throw new WalletNotFoundError();
+            if (!wallet.isBitKeep) throw new WalletNotInstalledError();
+
+            // @TODO: handle if popup is blocked
+
+            let account: string;
+            try {
+                account = await wallet.getAccount();
+            } catch (error: any) {
+                throw new WalletAccountError(error?.message, error);
+            }
+
+            let publicKey: PublicKey;
+            try {
+                publicKey = new PublicKey(account);
+            } catch (error: any) {
+                throw new WalletPublicKeyError(error?.message, error);
+            }
+
+            window.addEventListener('message', this._messaged);
+
+            this._wallet = wallet;
+            this._publicKey = publicKey;
+
+            this.emit('connect');
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        } finally {
+            this._connecting = false;
+        }
+    }
+
+    async disconnect(): Promise<void> {
+        if (this._wallet) {
+            window.removeEventListener('message', this._messaged);
+
+            this._wallet = null;
+            this._publicKey = null;
+
+            this.emit('disconnect');
+        }
+    }
+
+    async signTransaction(transaction: Transaction): Promise<Transaction> {
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            try {
+                return await wallet.signTransaction(transaction);
+            } catch (error: any) {
+                throw new WalletSignTransactionError(error?.message, error);
+            }
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        }
+    }
+
+    async signAllTransactions(transactions: Transaction[]): Promise<Transaction[]> {
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            try {
+                return await wallet.signAllTransactions(transactions);
+            } catch (error: any) {
+                throw new WalletSignTransactionError(error?.message, error);
+            }
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        }
+    }
+
+    private _messaged = (event: MessageEvent) => {
+        const data = event.data;
+        if (data && data.origin === 'bitkeep_internal' && data.type === 'lockStatusChanged' && !data.payload) {
+            this._disconnected();
+        }
```

Comment:
This appears to be copied from MathWallet. Unless BitKeep works by using a popup window in the same way as MathWallet, I would recommend basing this off a different adapter. See https://github.com/solana-labs/wallet-adapter/blob/master/packages/solflare/src/adapter.ts for how Solflare's adapter is structured.

Reply:
Ok, we will update as Solfare's adapter.